spec/lib/content_security_policy_spec.rb

```diff
@@ -78,14 +79,50 @@
         http://test.localhost/extra-locales/
       ])
     end
+  end
 
-    it 'can be extended with more sources' do
-      SiteSetting.content_security_policy_script_src = 'example.com|another.com'
-      script_srcs = parse(ContentSecurityPolicy.new.build)['script-src']
-      expect(script_srcs).to include('example.com')
-      expect(script_srcs).to include('another.com')
-      expect(script_srcs).to include("'unsafe-eval'")
-    end
+  it 'can be extended by plugins' do
+    plugin = Plugin::Instance.new(nil, "#{Rails.root}/spec/fixtures/plugins/csp_extension/plugin.rb")
+
+    plugin.activate!
+    Discourse.plugins << plugin
+    SiteSetting.load_settings("#{Rails.root}/spec/fixtures/plugins/csp_extension/config/settings.yml")
```

Comment:
Are you 100% sure state is not going to get leaked here between tests?

Reply:
Jinx ;) was just about to push the change